samples/sampleclient.js

```diff
@@ -33,15 +33,35 @@ if (fs.existsSync(keyPath)) {
   keys = keyFile.installed || keyFile.web;
 }
 
+const invalidRedirectUri = `The provided keyfile does not define a valid 
+redirect URI. There must be at least one redirect URI defined, and this sample
+assumes it redirects to 'http://localhost:3000/oauth2callback'.  Please edit 
+your keyfile, and add a 'redirect_uris' section.  For example:
+
+"redirect_uris": [
+  "http://localhost:3000/oauth2callback"
+]
+`;
+
 class SampleClient {
   constructor (options) {
     this._options = options || { scopes: [] };
 
+    // validate the redirectUri.  This is a frequent cause of confusion.
+    if (!keys.redirect_uris || keys.redirect_uris.length === 0) {
+      throw new Error(invalidRedirectUri);
+    }
+    const redirectUri = keys.redirect_uris[keys.redirect_uris.length - 1];
+    const parts = url.parse(redirectUri, false);
+    if (redirectUri.length > 0 && parts.port !== '3000' || parts.hostname !== 'localhost' || parts.path !== '/oauth2callback') {
```

Comment:
This seems a bit restrictive. Don't we just need to make sure there is a redirect URI?